### DISCUSSION TITLE
[Question] 与弹幕服务器的连接依然使用了代理

### DISCUSSION THREAD

**Discussion by Skimige** [HMBSbige/BilibiliLiveRecordDownLoader#45] [Q&A] [answered]
这是预期行为吗？网络 - 代理已设为“不使用代理”。
图示为我通过 CfW 手动掐断所有连接，日志反应确实走了 CfW 提供的系统代理。

![image](https://user-images.githubusercontent.com/9017470/106618828-df4ad080-65aa-11eb-8296-2266a8115956.png)
![image](https://user-images.githubusercontent.com/9017470/106618856-e96ccf00-65aa-11eb-910f-aaa6d0af9c20.png)

**Comment by HMBSbige** [comment]
不知道 CFW

设置不使用代理只是不会走系统代理，但如果是网络层被拦截的话确实也没什么办法

**Reply by Skimige** [reply]
CfW = Clash for Windows
hmmm… 我应该没用 TAP 之类的操作啊（

**Reply by Skimige** [reply]
![image](https://user-images.githubusercontent.com/9017470/116954891-b6c69580-acc3-11eb-876f-9d2e8254f105.png)

再丢个图吧（总之我也很迷惑

**Comment by SekiBetu** [comment]
https://docs.cfw.lbyczf.com/contents/tun.html#windows
Clash只有一个配置文件，所有的节点信息和模式都内嵌其中，注意查看是否开了TAP和TUN

**Reply by Skimige** [reply]
我从来没使用过 TUN 和 TAP 模式（
等我回去再确定一下。

**Reply by SekiBetu** [reply]
> 我从来没使用过 TUN 和 TAP 模式（
> 等我回去再确定一下。

https://github.com/ClashDotNetFramework/ClashDotNetFramework
https://github.com/ClashDotNet/ClashDotNet
偷偷安利下这俩新星，因为CFW是基于electron的，既吃内存又卡顿，这俩新项目刚刚建立，基于clash内核，用.NET框架制作的UI界面

**Reply by Skimige** [reply]
> https://github.com/ClashDotNetFramework/ClashDotNetFramework
> https://github.com/ClashDotNet/ClashDotNet
> 偷偷安利下这俩新星，因为CFW是基于electron的，既吃内存又卡顿，这俩新项目刚刚建立，基于clash内核，用.NET框架制作的UI界面

其中一个我用过，因为我现在的配置文件用了 providers，所以需要使用闭源内核（（
感谢安利，我后面尝试。

**Reply by SekiBetu** [reply]
> > [ClashDotNetFramework/ClashDotNetFramework](https://github.com/ClashDotNetFramework/ClashDotNetFramework)
> > [ClashDotNet/ClashDotNet](https://github.com/ClashDotNet/ClashDotNet)
> > 偷偷安利下这俩新星，因为CFW是基于electron的，既吃内存又卡顿，这俩新项目刚刚建立，基于clash内核，用.NET框架制作的UI界面
> 
> 其中一个我用过，因为我现在的配置文件用了 providers，所以需要使用闭源内核（（
> 感谢安利，我后面尝试。

这软件只是个UI，内核在bin目录自己替换即可

![image](https://user-images.githubusercontent.com/38283893/113505179-cd1bed00-956f-11eb-9959-57ac8b96285b.png)

**Comment by HMBSbige** [comment, marked as answer]
@Skimige 试试这个 [actions](https://github.com/HMBSbige/BilibiliLiveRecordDownLoader/actions/runs/849058018)

websocket 连接前也要手动设置代理（当然如果用 TCP 的弹幕服务器就完全不涉及到系统代理了）

~~我之前居然没注意到日志提示的是 websocket~~

**Reply by Skimige** [reply]
看起来修复了；
不过新版已经直接快进到 .NET 5.0 Runtime 了么（被引导去安装了

**Reply by HMBSbige** [reply]
一直都是 .NET5.0
跟 release 一样提供3个版本，不一定要装 runtime
